src/military/military.cpp

```diff
@@ -3959,6 +3964,33 @@ void cleanup_army(sys::state& state, dcon::army_id n) {
 	while(regs.begin() != regs.end()) {
 		state.world.delete_regiment((*regs.begin()).get_regiment());
 	}
+
+	auto b = state.world.army_get_battle_from_army_battle_participation(n);
+	if(b) {
+		bool should_end = true;
+		auto controller = state.world.army_get_controller_from_army_control(n);
+		if(bool(controller)) {
+			// TODO: Do they have to be in common war or can they just be "hostile against"?
+			for(auto bp : state.world.land_battle_get_army_battle_participation_as_battle(b)) {
+				if(bp.get_army() != n && are_allied_in_war(state, controller, state.world.army_get_controller_from_army_control(bp.get_army()))) {
+					should_end = false;
+				}
+			}
+		} else {
```

Comment:
You are still not handling the case where A and B are together in a battle fighting rebels. If you remove A, B should still fight the rebels even if there is no alliance between B and A